### PR TITLE
Add utilities for getting and checking tile operation parameters based on input parameters

### PR DESCRIPTION
**Pull request by msimberg**
Reduces duplication between the blas and cublas tile operations.

### CODE REVIEW

**Comment by msimberg** [comment]
borst try

File: include/dlaf/util_blas.h [hunk 1]
```diff
@@ -0,0 +1,146 @@
...
+//
+
+#pragma once
+
+#include <blas.hh>
+
+#include <dlaf/types.h>
+
+namespace dlaf {
+namespace util {
```
Comment (rasolca) [inline include/dlaf/util_blas.h:18]: I would put these methods in dlaf::tile::internal as I think that they are not part of the public API.

**Review by rasolca — APPROVED** [review]
LGTM, just a minor comment.

File: include/dlaf/util_blas.h (hunk 1 above)
Comment (msimberg) [reply include/dlaf/util_blas.h:18]: Yes, agreed! I'll change that.

**Comment by msimberg** [comment]
> borst try

That didn't quite go as planned.

bors try

File: include/dlaf/util_blas.h (hunk 1 above)
Comment (msimberg) [reply include/dlaf/util_blas.h]: Done.

**Review by albestro — APPROVED** [review]
I quickly went through and it LGTM.

Just one minor comment: in DLAF there are `Size2D` and `Index2D`, which I think they can be useful even for these things. They can be transposed and they accept a tag (for strong typing). In addition, they have allow few other manipulations (e.g. comparison and arithmetic operations).

**Pull request merged**